modules/tests/src/smithy4s/tests/PizzaClientSpec.scala

```diff
@@ -112,27 +115,59 @@ abstract class PizzaClientSpec extends IOSuite {
   )
 
   clientTestForError(
-    "Handle error w/o a discriminator header nor a unique status code",
-    Response(status = Status.ProxyAuthenticationRequired)
-      .withEntity(
-        Json.obj("message" -> Json.fromString("generic client error message"))
+    "Handle error  with a discriminator but can't be decoded",
+    Response[IO](status = Status.NotFound)
+      .withEntity("malformed body")
+      .withHeaders(Header.Raw(CIString("X-Error-Type"), "NotFoundError")),
+    rawErrorResponse(
+      404,
+      Map("X-Error-Type" -> "NotFoundError"),
+      "malformed body",
+      Some(
+        FailedDecodeAttempt(
+          discriminator = HttpDiscriminator.NameOnly("NotFoundError"),
+          contractError = HttpPayloadError(
+            path = smithy4s.codecs.PayloadPath(List()),
+            expected = "object",
+            message =
+              """Expected JSON object, offset: 0x00000000, buf:
+                |+----------+-------------------------------------------------+------------------+
+                ||          |  0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f | 0123456789abcdef |
+                |+----------+-------------------------------------------------+------------------+
+                || 00000000 | 6d 61 6c 66 6f 72 6d 65 64 20 62 6f 64 79       | malformed body   |
+                |+----------+-------------------------------------------------+------------------+""".stripMargin
+          )
```

Comment:
I really hate that this is part of the assertion. Let's avoid it : you could add an alternative to the definition of [clientTestForError](https://github.com/disneystreaming/smithy4s/blob/f3ca6635751fe85d29fec21059204f7917b5d28a/modules/tests/src/smithy4s/tests/PizzaClientSpec.scala#L219-L244) to abstract [this comparison away](https://github.com/disneystreaming/smithy4s/blob/f3ca6635751fe85d29fec21059204f7917b5d28a/modules/tests/src/smithy4s/tests/PizzaClientSpec.scala#L238), by doing : 

```scala
  def clientAssertError[E]( // feel free to find a better name
      name: String,
      response: Response[IO],
      assert: E => Unit
  )(implicit loc: SourceLocation) 
```

and have the current `clientTestForError` delegate to that one. Then your test could avoid comparing the error messages, as this is something that comes from an upstream library and really not something that should be tested here.